source-firestore/main.go

```diff
@@ -0,0 +1,411 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	firestore "cloud.google.com/go/firestore"
+	firebase "firebase.google.com/go"
+	schemagen "github.com/estuary/connectors/go-schema-gen"
+	"github.com/estuary/connectors/schema_inference"
+	"github.com/estuary/flow/go/protocols/airbyte"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+type State struct {
+	// Last time a full table scan was run, we use this to know if we need to do a full scan
+	// to reach eventual consistency
+	LastScan string
```

Comment:
Could this be a `time.Time` and use the default JSON serialization/deserialization logic?

Reply:
I thought about that, I just was not sure what the default value is in case of `time.Time` if it's not set? that's why I went with string and parsing it, and then setting the default to unix time 0

Reply:
Apparently it is the zero time, so I guess it works! https://stackoverflow.com/questions/23051973/what-is-the-zero-value-for-time-time-in-go